File: leet_0392_subsequence.js
var isSubsequence = function(s, t) {
    let len = s.length

    if (len == 0)
        return true
    if (t.length == 0)
        return false

    let i = 0
    for (let c of t){
        if (c == s[i])
            i++
        if (i == len)
            return true
    }
    return false
};

File: leet_0392_subsequence.go
func isSubsequence(s string, t string) bool {
    if s == "" {
        return true
    }
    i, j := 0, 0
    ls, lt := len(s), len(t)
    for i < ls && j < lt {
        if s[i] == t[j] {
            i++
        }
        if i == ls {
            return true
        }
        j++
    }
    return false
}
